blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubServerContainer.java

```diff
@@ -90,14 +93,23 @@
         }
 
         if (errors.isEmpty()) {
-            GitHubConfiguration config = GitHubConfiguration.get();
-            String sanitizedUrl = discardQueryString(url);
-            Endpoint endpoint = new Endpoint(sanitizedUrl, name);
-            if (!config.addEndpoint(endpoint)) {
-                errors.add(new ErrorMessage.Error(GithubServer.API_URL, ErrorMessage.Error.ErrorCodes.ALREADY_EXISTS.toString(), GithubServer.API_URL + " is already registered as '" + endpoint.getName() + "'"));
-            }
-            else {
-                return new GithubServer(endpoint, getLink());
+            SecurityContext old = null;
+            try {
+                // We need to escalate privilege to add user defined endpoint to
+                old = ACL.impersonate(ACL.SYSTEM);
+                GitHubConfiguration config = GitHubConfiguration.get();
+                String sanitizedUrl = discardQueryString(url);
```

Comment:
Why are we discarding query string? Maybe use `GithubConfiguration.normalizeApiUri()` for normalizing?